draco/analysis/flagging.py

```diff
@@ -697,9 +697,210 @@ def process(self, stream):
             "%0.5f%% of data is below the weight threshold" % (100.0 * (drop_frac))
         )
 
-        out.mask[:] = mask
+        mask_cont.mask[:] = mask
 
-        return out
+        return mask_cont
+
+
+class ThresholdVisWeightBaseline(task.SingleTask):
+    """Form a mask corresponding to weights that are below some threshold.
+
+    The threshold is determined as `maximum(absolute_threshold,
+    relative_threshold * average(weight))` and is evaluated per product/stack
+    entry. The user can specify whether to use a mean or median as the average,
+    but note that the mean is much more likely to be biased by anomalously
+    high- or low-weight samples (both of which are present in raw CHIME data).
+    The user can also specify that weights below some threshold should not be
+    considered when taking the average and constructing the mask (the default
+    is to only ignore zero-weight samples).
+
+    The task outputs a BaselineMask or SiderealBaselineMask depending on the
+    input container.
+
+    Parameters
+    ----------
+    average_type : string, optional
+        Type of average to use ("median" or "mean"). Default: "median".
+    absolute_threshold : float, optional
+        Any weights with values less than this number will be set to zero.
+        Default: 1e-7.
+    relative_threshold : float, optional
+        Any weights with values less than this number times the average weight
+        will be set to zero. Default: 1e-6.
+    ignore_absolute_threshold : float, optional
+        Any weights will values less than this number will be ignored when
+        taking averages and constructing the mask. Default: 0.0.
+    pols_to_flag : string, optional
+        Which polarizations to flag. "copol" only flags XX and YY baselines,
+        while "all" flags everything. Default: "all".
+    """
+
+    average_type = config.enum(["median", "mean"], default="median")
+    absolute_threshold = config.Property(proptype=float, default=1e-7)
+    relative_threshold = config.Property(proptype=float, default=1e-6)
+    ignore_absolute_threshold = config.Property(proptype=float, default=0.0)
+    pols_to_flag = config.enum(["all", "copol"], default="all")
+
+    def setup(self, telescope):
+        """Set the telescope model.
+
+        Parameters
+        ----------
+        telescope : TransitTelescope
+        """
+
+        self.telescope = io.get_telescope(telescope)
+
+    def process(
+        self,
+        stream,
+    ) -> Union[containers.BaselineMask, containers.SiderealBaselineMask]:
+        """Construct mask and apply to input weights, if desired.
```

Comment:
This task used to have the option to apply the mask to the input weights, but it doesn't any more
```suggestion
        """Construct baseline-dependent mask.
```